analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -3,617 +3,391 @@
+
+        var stack = new Stack<int>();
+        stack.Push(5);
+        stack.Clear();                  // Compliant
+
+        var obs = new ObservableCollection<int>();
+        obs.Add(5);
+        obs.Clear();                    // Compliant
+        obs = new ObservableCollection<int>();
+        obs.Insert(0, 5);
+        obs.Clear();                    // Compliant
+
+        var dict = new Dictionary<int, int>();
+        dict.Add(1, 5);
+        dict.Clear();                   // Compliant
     }
 }
 
-class ObservableCollectionTests
+class AdvancedTests
 {
-    private IEnumerable<int> items = new ObservableCollection<int> { 1, 2, 3 };
-
-    private static bool Predicate(int i) => true;
-    private static void Action(int i) { }
-    private ObservableCollection<int> Factory() => new ObservableCollection<int>();
-
-    public void DefaultConstructor_Applies_Empty()
-    {
-        var list = new ObservableCollection<int>();
-        list.Clear(); // FIXME Non-compliant
-    }
-
-    public void ConstructorWithEnumerable_Applies_NonEmpty()
-    {
-        var list = new ObservableCollection<int>(items);
-        list.Clear(); // Compliant
-    }
-
-    public void ConstructorWithInitializer_Applies_NonEmpty()
-    {
-        var list = new ObservableCollection<int> { 1, 2, 3 };
-        list.Clear(); // Compliant
-    }
-
-    public void ConstructorWithEmptyInitializer_Applies_Empty()
-    {
-        var list = new ObservableCollection<int> { };
-        list.Clear(); // FIXME Non-compliant
-    }
-
-    public void Other_Initialization_Applies_No_Constraints()
-    {
-        var list = Factory();
-        list.Clear(); // Compliant, we don't know anything about the list
-    }
-
-    public void Methods_Raise_Issue()
-    {
-        var list = new ObservableCollection<int>();
-        list.Clear(); // FIXME Non-compliant
-        list.IndexOf(1); // FIXME Non-compliant
-        list.Move(1, 0); // FIXME Non-compliant
-        list.Remove(1); // FIXME Non-compliant
-        var x = list[5]; // FIXME Non-compliant
-        list[5] = 5; // FIXME Non-compliant
-    }
-
-    public void Methods_Ignored()
-    {
-        var list = new ObservableCollection<int>();
-        list.GetHashCode();
-        list.Equals(items);
-        list.GetType();
-        list.ToString();
-        var a = list.Count;
-        list.CollectionChanged += List_CollectionChanged;
-    }
-
-    private void List_CollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
-    {
-        throw new NotImplementedException();
-    }
-
-    public void Methods_Set_NotEmpty()
+    public void ExtensionMethods_Should_Not_Raise()
     {
-        var list = new ObservableCollection<int>();
-        list.Add(1);
-        list.Clear(); // Compliant, will normally raise
-
-        list = new ObservableCollection<int>();
-        list.Insert(0, 1);
-        list.Clear(); // Compliant
+        var list = new List<int>();
+        list.Any();     // Compliant
```

Comment:
I'd consider this one (and similar friends) FN. While `.CustomExtension()` as compliant